libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -185,8 +492,14 @@ function not(x1, x2, x3, x4) -> r1, r2, r3, r4 {
 	let mask := 0xffffffffffffffff
 	r1, r2, r3, r4 := xor(x1, x2, x3, x4, mask, mask, mask, mask)
 }
-function iszero(x1, x2, x3, x4) -> r1, r2, r3, r4 {
-	r4 := i64.eqz(i64.or(i64.or(x1, x2), i64.or(x3, x4)))
+function iszero(x1, x2, x3, x4) -> r {
+	r := i64.eqz(i64.or(i64.or(x1, x2), i64.or(x3, x4)))
+}
+function iszero320(x1, x2, x3, x4, x5) -> r {
+	r := i64.eqz(i64.or(i64.or(i64.or(x1, x2), i64.or(x3, x4)), x5))
+}
+function iszero512(x1, x2, x3, x4, x5, x6, x7, x8) -> r {
+	r := i64.eqz(i64.or(i64.or(i64.or(i64.or(i64.or(i64.or(x1, x2), i64.or(x3, x4)), x5), x6), x7), x8))
```

Comment:
Wouldn't it be better to re-group them to get a complete binary tree?

Reply:
Or just combine `iszero(x1, x2, x3, x4)` and `iszero(x5, x6, x7, x8)`?

Reply:
Agreed. Will use second option.